spec/controllers/api/v1/rents_controller_spec.rb

```diff
@@ -0,0 +1,80 @@
+require 'rails_helper'
+
+describe RentsController, type: :controller do
+  describe 'GET #index' do
+    context 'without authenticated user' do
+      subject!(:http_request) { get :index }
+
+      let!(:rents) { create_list(:rent, 5) }
+
+      it 'responds with status 401 (UNAUTHORIZED)' do
+        expect(response).to have_http_status(:unauthorized)
+      end
+    end
+
+    context 'when fetching all the rents' do
+      include_context 'with authenticated user'
+      subject!(:http_request) { get :index }
+
+      let!(:rents) { create_list(:rent, 5) }
+
+      it 'responds with the books JSON' do
+        expected = ActiveModel::Serializer::CollectionSerializer.new(
+          rents, each_serializer: RentSerializer
+        ).to_json
+        expect(response.body.to_json) =~ JSON.parse(expected)
+      end
+
+      it 'responds with status 200 (OK)' do
+        expect(response).to have_http_status(:ok)
+      end
+    end
+
+    context 'without any rent' do
+      include_context 'with authenticated user'
+      subject!(:http_request) { get :index }
+
+      it 'responds with an empty body' do
+        expect(JSON.parse(response.body)['page']).to be_empty
+      end
+
+      it 'responds with status 200 (OK)' do
+        expect(response).to have_http_status(:ok)
+      end
+    end
+  end
+
+  describe 'POST #create' do
+    context 'without authenticated user' do
+      it 'responds with status 401 (UNAUTHORIZED)' do
+        post :create, params: FactoryBot.attributes_for(:rent)
```

Comment:
before { post :create, params: FactoryBot.attributes_for(:rent) }

Reply:
I didn't understand this correction. Why do we add the before?

Reply:
It's to keep easier the read of  the example
Before
```rb
     it 'responds with status 401 (UNAUTHORIZED)' do
        post :create, params: FactoryBot.attributes_for(:rent)
        expect(response).to have_http_status(:unauthorized)
      end
```
After, note the test should only need one line
```rb
    before { post :create, params: FactoryBot.attributes_for(:rent) }

     it 'responds with status 401 (UNAUTHORIZED)' do
        expect(response).to have_http_status(:unauthorized)
      end
```

Reply:
Okey!